src/hooks/api/useAccounts.ts

```diff
@@ -6,20 +6,44 @@ import {
 import { Account } from '@/book/entities';
 import fetcher from './fetcher';
 
-export function useAccount(guid: string): UseQueryResult<Account> {
-  const result = useQuery({
-    queryKey: [...Account.CACHE_KEY, { guid }],
-    queryFn: fetcher(() => Account.findOneBy({ guid }), `/${Account.CACHE_KEY.join('/')}/${guid}`),
-  });
+export function useAccount(guid: string): UseQueryResult<Account | undefined> {
+  const result = useAccounts<Account | undefined>(
+    (data => data.find(a => a.guid === guid)),
+  );
 
   return result;
 }
 
-export function useAccounts(): UseQueryResult<Account[]> {
+export function useAccounts<TData = Account[]>(
+  select?: (data: Account[]) => TData,
+): UseQueryResult<TData> {
   const result = useQuery({
     queryKey: [...Account.CACHE_KEY],
-    queryFn: fetcher(() => Account.find(), `/${Account.CACHE_KEY.join('/')}`),
+    queryFn: fetcher(
+      async () => {
+        const accounts = await Account.find();
+        const root = accounts.find(a => a.type === 'ROOT') as Account;
+        setAccountPaths(root, accounts);
```

Comment:
every time we re-fetch the accounts, we calculate the paths. Yes, it's true there are chances we re-calculate the same but for now leaving as is, we can always optimise